mason.sh

```diff
@@ -505,7 +505,8 @@ function mason_config {
 function mason_write_config {
     local INI_FILE
     INI_FILE="${MASON_PREFIX}/mason.ini"
-    echo "`mason_config`" > "${INI_FILE}"
+    MASON_INI_DATA=$(set -e;mason_config)
+    echo "${MASON_INI_DATA}" > "${INI_FILE}"
```

Comment:
If line 508 fails, now the `set- e` will prevent any blank mason.ini from being saved. In local testing it seemed this way needed.